lute/parse/space_delimited_parser.py

```diff
@@ -25,6 +30,142 @@ class SpaceDelimitedParser(AbstractParser):
     def name(cls):
         return "Space Delimited"
 
+    @staticmethod
+    @functools.lru_cache
+    def compile_re_pattern(pattern: str, *args, **kwargs) -> re.Pattern:
+        """Compile regular expression pattern, cache result for fast re-use."""
+        return re.compile(pattern, *args, **kwargs)
+
+    @staticmethod
+    @functools.lru_cache
+    def get_default_word_characters() -> str:
+        """Return default value for lang.word_characters."""
+
+        # Unicode categories reference: https://www.compart.com/en/unicode/category
+        categories = set(["Cf", "Ll", "Lm", "Lo", "Lt", "Lu", "Mc", "Mn", "Sk"])
+
+        # There are more than 130,000 characters across all these categories.
+        # Expressing this a single character at a time, mostly using unicode
+        # escape sequences like \u1234 or \U12345678, would require 1 megabyte.
+        # Converting to ranges like \u1234-\u1256 requires only 10K.
+        ranges = []
+        current = None
+        for i in range(1, sys.maxunicode):
+            c = chr(i)
+            category = unicodedata.category(c)
+
+            if category not in categories:
+                continue
+
+            if current is None:
+                current = [i, i]
+                continue
+
+            if current[1] == (i - 1):
+                current[1] = i
+                continue
+
+            s1 = (r"\u{:04x}" if current[0] < 0x10000 else r"\U{:08x}").format(
+                current[0]
+            )
+            if current[0] == current[1]:
+                ranges.append(s1)
+            else:
+                s2 = (r"\u{:04x}" if current[1] < 0x10000 else r"\U{:08x}").format(
+                    current[1]
+                )
+                ranges.append(s1 + "-" + s2)
+            current = [i, i]
+
+        if current is not None:
```

Comment:
This section adds the last range to the list of ranges.  `current` is never reset to `None` in the above, so after the iteration, this will always be called.

Lines 81-90 are identical to 68-77.